src/plugins/duplicate-finder/index.ts

```diff
@@ -0,0 +1,456 @@
+              },
+              {
+                label: 'Similarity Threshold',
+                submenu: [70, 80, 85, 90, 95].map((threshold) => ({
+                  label: `${threshold}%`,
+                  type: 'radio',
+                  checked: config.similarityThreshold === threshold,
+                  click() {
+                    setConfig({ similarityThreshold: threshold });
+                  },
+                })),
+              },
+            ],
+          },
+          {
+            type: 'separator',
+          },
+          {
+            label: 'Show Results Panel',
+            click() {
+              window.postMessage({ type: 'DUPLICATE_FINDER_SHOW_PANEL' }, '*');
+            },
+          },
+        ],
+      },
+    ];
+  },
+
+  backend: {
+    start({ ipc }) {
+      // Handle scan request from renderer
+      ipc.handle('duplicate-finder-scan', async (scanOptions) => {
+        return { success: true, message: 'Scan initiated' };
+      });
+
+      // Handle duplicate removal
+      ipc.handle('duplicate-finder-remove', async (duplicateIds) => {
+        return { success: true, removed: duplicateIds.length };
+      });
+    },
+
+    onConfigChange(newConfig) {
+      console.log('Duplicate Finder config changed:', newConfig);
+    },
+
+    stop(context) {
+      // Cleanup when plugin is disabled
+    },
+  },
+
+  renderer: {
+    async start(context) {
+      let duplicateGroups: DuplicateGroup[] = [];
+      let isScanning = false;
+
+      // Utility functions
+      const normalizeString = (str: string): string => {
+        return str
+          .toLowerCase()
+          .replace(/[^\w\s]/g, '')
+          .replace(/\s+/g, ' ')
+          .trim();
+      };
+
+      const calculateSimilarity = (str1: string, str2: string): number => {
+        const normalized1 = normalizeString(str1);
+        const normalized2 = normalizeString(str2);
+        
+        if (normalized1 === normalized2) return 100;
+        
+        // Levenshtein distance based similarity
+        const matrix = Array(normalized2.length + 1).fill(null).map(() => 
+          Array(normalized1.length + 1).fill(null)
+        );
+        
+        for (let i = 0; i <= normalized1.length; i++) matrix[0][i] = i;
+        for (let j = 0; j <= normalized2.length; j++) matrix[j][0] = j;
+        
+        for (let j = 1; j <= normalized2.length; j++) {
+          for (let i = 1; i <= normalized1.length; i++) {
+            const indicator = normalized1[i - 1] === normalized2[j - 1] ? 0 : 1;
+            matrix[j][i] = Math.min(
+              matrix[j][i - 1] + 1,
+              matrix[j - 1][i] + 1,
+              matrix[j - 1][i - 1] + indicator
+            );
+          }
+        }
+        
+        const distance = matrix[normalized2.length][normalized1.length];
+        const maxLength = Math.max(normalized1.length, normalized2.length);
+        return Math.round(((maxLength - distance) / maxLength) * 100);
+      };
```

Comment:
we already have a library for string similarity, although it's a different algorithm

```typescript
import { jaroWinkler } from '@skyra/jaro-winkler';
```